tests/components/energyzero/test_services.py

```diff
@@ -42,14 +39,38 @@ async def test_service(
 
     data = incl_vat | start | end
 
-    try:
-        response = await hass.services.async_call(
+    assert snapshot == await hass.services.async_call(
+        DOMAIN,
+        service,
+        data,
+        blocking=True,
+        return_response=True,
+    )
+
+
+@pytest.mark.usefixtures("init_integration")
+@pytest.mark.parametrize("service", [GAS_SERVICE_NAME, ENERGY_SERVICE_NAME])
+@pytest.mark.parametrize(
+    "service_data",
+    [
+        {},
+        {"incl_vat": "incorrect vat"},
+        {"incl_vat": True, "start": "incorrect date"},
+        {"incl_vat": True, "end": "incorrect date"},
+    ],
+)
+async def test_service_validation(
+    hass: HomeAssistant,
+    service: str,
+    service_data: dict[str, str],
+) -> None:
+    """Test the EnergyZero Service validation."""
+
+    with pytest.raises((ServiceValidationError, vol.er.Error)):
```

Comment:
Please pass the expected error as argument and match to the correct message of it